app/models/user_organization.rb

```diff
@@ -10,11 +10,28 @@ class UserOrganization < ActiveRecord::Base
   belongs_to :organization, inverse_of: :user_organizations
 
   before_destroy :destroy_associations
+  after_create :create_samples_table_state
 
   def role_str
     I18n.t("user_organizations.enums.role.#{role.to_s}")
   end
 
+  def create_samples_table_state
+    org_status = SampleDatatable::SAMPLES_TABLE_DEFAULT_STATE.deep_dup
+    organization.custom_fields.each_with_index do |_, index|
+      org_status['columns'] << { 'visible' => true,
+                                 'search' => { 'search' => '',
+                                               'smart' => true,
+                                               'regex' => false,
+                                               'caseInsensitive' => true } }
```

Comment:
Can't you just use SAMPLES_TABLE_DEFAULT_STATE['columns'] directly, without org_status?